### PR TITLE
feat(lint): add --compact flag for terse output

### PR DESCRIPTION
**Pull request by brenelz**
Related to https://github.com/denoland/deno_lint/issues/1077

Added a --compact linting flag similar to the --json flag

Screen Shot 2022-09-16 at 6 59 08 PM

### CODE REVIEW

File: cli/args/flags.rs [hunk 1]
```diff
@@ -1386,6 +1387,12 @@ Ignore linting a file by adding an ignore comment at the top of the file:
         .help("Output lint result in JSON format")
         .takes_value(false),
     )
+    .arg(
+      Arg::new("compact")
+        .long("compact")
+        .help("Output lint result in compact format")
+        .takes_value(false),
```
Comment (bartlomieju) [inline cli/args/flags.rs:1394]: This arg should have `conflicts_with()` setting, so users can't use `--compat` with `--json`

**Review by bartlomieju — COMMENTED** [review]
Looks good! Please add an integration test in `cli/tests/integration_tests/lint_tests.rs` that exercises `--compact` flag.

File: cli/args/flags.rs (hunk 1 above)
Comment (brenelz) [reply cli/args/flags.rs]: good catch. Its been updated

File: cli/tools/lint.rs [hunk 2]
```diff
@@ -413,6 +418,50 @@ impl LintReporter for PrettyLintReporter {
...
+    CompactLintReporter { lint_count: 0 }
+  }
+}
+
+impl LintReporter for CompactLintReporter {
+  fn visit_diagnostic(&mut self, d: &LintDiagnostic, _source_lines: Vec<&str>) {
+    self.lint_count += 1;
+
+    eprintln!(
+      "{}: line {}, col {}, Error - {} ({})",
```
Comment (bartlomieju) [inline cli/tools/lint.rs:436]:
```suggestion
      "{}: line {}, col {} - {} ({})",
```

**Review by bartlomieju — COMMENTED** [review]
Last suggestions, otherwise I think this is good to land

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks @brenelz! I think we should follow up with adding a setting to `deno.json` that allows to set preferred report, something like:
```
{
  "lint": {
    "report": "compact"
  }
}
```
Is it something you'd be willing to look into?

**Comment by brenelz** [comment]
@bartlomieju yeah I can try and do that. Basically if you don't pass --json or --compact it would use the default from deno.json?

**Comment by bartlomieju** [comment]
> @bartlomieju yeah I can try and do that. Basically if you don't pass --json or --compact it would use the default from deno.json?

Yes, exactly that!

**Pull request merged**